File: src/main/java/net/dontdrinkandroot/gitki/wicket/model/FilePathStringContentModel.java
package net.dontdrinkandroot.gitki.wicket.model;

import net.dontdrinkandroot.gitki.model.FilePath;
import net.dontdrinkandroot.gitki.service.GitService;
import net.dontdrinkandroot.wicket.model.AbstractChainedInjectedLoadableDetachableModel;
import org.apache.wicket.WicketRuntimeException;
import org.apache.wicket.model.IModel;

import javax.inject.Inject;
import java.io.IOException;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class FilePathStringContentModel extends AbstractChainedInjectedLoadableDetachableModel<FilePath, String>
{
    @Inject
    private GitService gitService;

    public FilePathStringContentModel(IModel<? extends FilePath> parentModel)
    {
        super(parentModel);
    }

    @Override
    protected String load()
    {
        try {
            return this.gitService.getContentAsString(this.getParentObject().toPath());
        } catch (IOException e) {
            throw new WicketRuntimeException(e);
        }
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/model/FilePathStringContentModel.kt
package net.dontdrinkandroot.gitki.wicket.model

import net.dontdrinkandroot.gitki.model.FilePath
import net.dontdrinkandroot.gitki.service.git.GitService
import net.dontdrinkandroot.wicket.model.AbstractChainedInjectedLoadableDetachableModel
import org.apache.wicket.WicketRuntimeException
import org.apache.wicket.model.IModel
import java.io.IOException
import javax.inject.Inject

class FilePathStringContentModel(parentModel: IModel<out FilePath>) :
    AbstractChainedInjectedLoadableDetachableModel<FilePath, String>(parentModel) {

    @Inject
    private lateinit var gitService: GitService

    override fun load() = try {
        gitService.getContentAsString(this.parentObject!!)
    } catch (e: IOException) {
        throw WicketRuntimeException(e)
    }
}